core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/allowances/AbstractAllowanceWithRanges.java

```diff
@@ -190,10 +190,13 @@ private Envelope[] computeAllowanceRegion(Envelope envelopeRegion, EnvelopeSimCo
             var envelopeRange = Envelope.make(envelopeRegion.slice(range.beginPos, range.endPos));
             var imposedBeginSpeed = imposedTransitionSpeeds[rangeIndex];
             var imposedEndSpeed = imposedTransitionSpeeds[rangeIndex + 1];
+            var tolerance = context.timeStep * envelopeRange.getTotalTime() / envelopeRegion.getTotalTime();
```

Comment:
Could we create this intermediate variable to make a parallel with the sectionRatio below ?
```suggestion
            var rangeRatio = envelopeRange.getTotalTime() / envelopeRegion.getTotalTime()
            var tolerance = context.timeStep * rangeRatio;
```